tests/playwright-test/playwright.ct-react.spec.ts

```diff
@@ -250,6 +250,57 @@ test('should work with JSX in variable', async ({ runInlineTest }) => {
   expect(result.passed).toBe(1);
 });
 
+test('should pass "key" attribute from JSX in variable', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'playwright.config.ts': playwrightCtConfigText,
+    'playwright/index.html': `<script type="module" src="./index.js"></script>`,
+    'playwright/index.js': `
+    `,
+
+    'src/container.jsx': `
+      import { useState } from 'react';
+      export function Container({ children }) {
+        const [index, setIndex] = useState(0);
+        return (
+          <div onClick={() => setIndex((index + 1) % children.length)}>
+            {children[index]}
+          </div>
+        );
+      }
+    `,
+
+    'src/button.jsx': `
+      import { useState } from 'react';
+      export function Button({ value }) {
+        const [state, setState] = useState(value);
+        return <button onClick={() => setState(state + 1)}>{state}</button>;
+      }
+    `,
+
+    'src/index.test.jsx': `
+      import { test, expect } from '@playwright/experimental-ct-react';
+      import { Button } from './button';
+      import { Container } from './container';
+
+      test('key should tear down and recreate component', async ({ mount }) => {
+        const component = await mount(
+          <Container>
+            <Button key='a' value={5} />
```

Comment:
Let's make the test more clear by using different values in this two buttons, like 1 and 10, so that the expectations below are more obvious.

Reply:
updated